File: MilkManagementproject/milk_system/src/main/java/com/example/services/CustomerService.java
package com.example.services;

import java.util.List;
import java.util.Optional;

import com.example.model.Customer;


public interface CustomerService {
	
	
	public List<Customer>getAllCustomers() ;
	
	public Customer addCustomer(Customer customer);
	
	public Optional<Customer> getCustomerById(Long customerId);
	
	public void deleteCustomer(Long customerId);
	
	public Customer updateCustomer(Long customerId, Customer updatedCustomer);
    }

File: MilkManagementproject/Frontend_milk/milk_system/src/services/customerService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/customers";

export const getAllCustomers = () => axios.get(API_URL);
export const addCustomer = (customer) => axios.post(API_URL, customer);
export const getCustomerById = (id) => axios.get(`${API_URL}/${id}`);
export const updateCustomer = (id, customer) => axios.put(`${API_URL}/${id}`, customer);
export const deleteCustomer = (id) => axios.delete(`${API_URL}/${id}`);